portal/flatpak-portal.c

```diff
@@ -844,6 +1064,35 @@ handle_spawn (PortalFlatpak         *object,
       g_ptr_array_add (flatpak_argv, g_strdup ("--parent-expose-pids"));
     }
 
+  notify_start = (arg_flags & FLATPAK_SPAWN_FLAGS_NOTIFY_START) != 0;
+  if (notify_start)
+    {
+      int fds[2];
+      if (pipe (fds) == -1)
+        {
+          int errsv = errno;
+          g_dbus_method_invocation_return_error (invocation, G_IO_ERROR,
+                                                 g_io_error_from_errno (errsv),
+                                                 "Failed to create instance ID pipe: %s",
+                                                 g_strerror (errsv));
+          return TRUE;
+        }
+
+      GInputStream *in_stream = G_INPUT_STREAM (g_unix_input_stream_new (fds[0], TRUE));
+      /* This is saved to ensure the portal's end gets closed after the exec. */
+      instance_id_out_stream = G_OUTPUT_STREAM (g_unix_output_stream_new (fds[1], TRUE));
+
+      instance_id_read_data = g_new0 (InstanceIdReadData, 1);
+      instance_id_read_data->buffer = g_malloc0 (INSTANCE_ID_BUFFER_SIZE);
```

Comment:
Why don't you just put the buffer inline in the InstanceIdReadData struct? No need for a separate allocation.